cocoa/BugsnagReactNative.m

```diff
@@ -296,6 +308,63 @@ + (void)startWithConfiguration:(BugsnagConfiguration *)config {
         // session to compensate.
         [Bugsnag resumeSession];
     }
+    [self addRuntimeVersionToMetaData:config];
+}
+
+/**
+ * Stores runtime version info in a metadata tab (it will be moved to a
+ * different payload location before sending)
+ */
+- (void)addRuntimeVersionToMetaData:(BugsnagConfiguration *)config {
+    [config.metaData addAttribute:@"reactNativeVersion"
+                        withValue:[self findReactNativeVersion]
+                    toTabWithName:@"_bugsnag"];
+}
+
+- (NSDictionary *)removeRuntimeVersionFromMetaData:(BugsnagCrashReport *)report {
+    NSMutableDictionary *metadata = report.metaData.mutableCopy;
+    metadata[@"_bugsnag"] = nil;
+    return metadata;
+}
+
+- (NSDictionary *)addDeviceRuntimeVersion:(NSDictionary *)device
+                       reactNativeVersion:(NSString *)reactNativeVersion {
+    NSMutableDictionary *copy = [device mutableCopy];
+    NSMutableDictionary *runtimeVersions = [copy[@"runtimeVersions"] mutableCopy];
+
+    if (runtimeVersions == nil) {
+        runtimeVersions = [NSMutableDictionary new];
+    }
+    runtimeVersions[@"reactNative"] = reactNativeVersion;
+    copy[@"runtimeVersions"] = runtimeVersions;
+    return copy;
+}
+
+// see https://github.com/facebook/react-native/blob/6df2edeb2a33d529e4b13a5b6767f300d08aeb0a/scripts/bump-oss-version.js
+- (NSString *)findReactNativeVersion {
+    NSDictionary *versionMap = RCTGetReactNativeVersion();
```

Comment:
This call should be wrapped so it only has to happen once and is cached after that.

```objc
- (NSString *)findReactNativeVersion {
    static dispatch_once_t onceToken;
    static NSString *BSGReactNativeVersion = nil;
    dispatch_once(&onceToken, ^{
        NSDictionary *versionMap = RCTGetReactNativeVersion();
        NSNumber *major = versionMap[@"major"];
        NSNumber *minor = versionMap[@"minor"];
        NSNumber *patch = versionMap[@"patch"];
        NSString *prerelease = versionMap[@"prerelease"];
        NSMutableString *versionString = [NSMutableString new];

        if (![major isEqual:[NSNull null]]) {
            [versionString appendString:[major stringValue]];
            [versionString appendString:@"."];
        }
        if (![minor isEqual:[NSNull null]]) {
            [versionString appendString:[minor stringValue]];
            [versionString appendString:@"."];
        }
        if (![patch isEqual:[NSNull null]]) {
            [versionString appendString:[patch stringValue]];
        }
        if (![prerelease isEqual:[NSNull null]]) {
            [versionString appendString:@"-"];
            [versionString appendString:prerelease];
        }
        BSGReactNativeVersion = [NSString stringWithString:versionString];
    });
    return BSGReactNativeVersion;
}
```

Ref: [`dispatch_once`](https://developer.apple.com/documentation/dispatch/1447169-dispatch_once)